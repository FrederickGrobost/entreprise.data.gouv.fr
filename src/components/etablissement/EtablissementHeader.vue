<template>
  <div class="company">
    <div class="company__main">
      <header-skeleton v-if="isEtablissementLoading"></header-skeleton>
      <div v-else class="title__block">
        <h2 v-if="haveSireneInfo">
          <div v-if="isEntrepreneurIndividuel">
            {{
              concatNames(
                resultSirene.unite_legale.prenom_1,
                resultSirene.unite_legale.nom
              ) | ifExist
            }}
          </div>
          <div v-else>
            {{ resultSirene.unite_legale.denomination | removeExtraChars }}
          </div>
          <span class="company__siren"
            >(<span
              :inner-html.prop="resultSirene.siren | prettySirenHtml"
            />)</span
          >
        </h2>
        <h2 v-if="haveOnlyRNAInfo">
          {{ resultRNA.titre }}
          <span class="association__id">({{ resultRNA.id_association }})</span>
        </h2>

        <template v-if="haveSireneInfo">
          <div class="subtitle">
            <div>{{ resultSirene.geo_l4 }}</div>
            <div>{{ resultSirene.geo_l5 }}</div>
          </div>
          <div class="second__subtitle">
            {{ resultSirene.libelle_activite_principale_entreprise }}
          </div>
        </template>
        <div v-if="haveOnlyRNAInfo" class="second__subtitle">
          {{ resultRNA.titre_court }}
        </div>
        <etablissement-header-sirene-children v-if="haveSireneInfo" />

        <router-link
          v-if="haveSireneInfo"
          :to="{ name: 'RNCS', params: { searchId: resultSirene.siren } }"
        >
          Fiche d'immatriculation au RNCS
        </router-link>
        <a v-if="haveSireneInfo" class="link__external" :href="linkBodacc"
          >Liste des annonces BODACC</a
        >
        <etablissement-header-timestamp
          :result-sirene="resultSirene"
          :result-r-n-a="resultRNA"
        />
      </div>
      <div v-if="isEtablissementLoading" class="map__dummy panel"></div>
      <template v-else>
        <etablissement-map
          v-if="haveSireneInfo"
          :position-etablissement="coordinates"
          :etablissement="resultSirene"
        />
      </template>
    </div>
  </div>
</template>

<script>
import Filters from "@/components/mixins/filters";
import Formating from "@/components/mixins/formating";
import EtablissementHeaderSireneChildren from "@/components/etablissement/etablissementHeader/EtablissementHeaderSireneChildren";
import EtablissementMap from "@/components/etablissement/EtablissementMap";
import HeaderSkeleton from "@/components/etablissement/skeletons/HeaderSkeleton";
import EtablissementHeaderTimestamp from "@/components/etablissement/etablissementHeader/EtablissementHeaderTimestamp";

export default {
  name: "EtablissementHeader",
  components: {
    EtablissementHeaderTimestamp: EtablissementHeaderTimestamp,
    EtablissementHeaderSireneChildren: EtablissementHeaderSireneChildren,
    EtablissementMap: EtablissementMap,
    HeaderSkeleton: HeaderSkeleton
  },
  mixins: [Filters, Formating],
  props: { searchId: { type: String, default: "" } },
  computed: {
    isEtablissementLoading() {
      return this.$store.getters.mainAPISLoading;
    },
    resultSirene() {
      return this.$store.getters.singlePageEtablissementSirene;
    },
    resultRNA() {
      return this.$store.getters.singlePageEtablissementRNA;
    },
    haveOnlyRNAInfo() {
      return !this.haveSireneInfo && this.haveRNAInfo;
    },
    haveSireneInfo() {
      if (this.$store.getters.sireneAvailable) {
        return true;
      }
      return false;
    },
    haveRNAInfo() {
      if (this.$store.getters.RNAAvailable) {
        return true;
      }
      return false;
    },
    coordinates() {
      if (
        this.resultSirene &&
        this.resultSirene.longitude &&
        this.resultSirene.latitude
      ) {
        return [this.resultSirene.longitude, this.resultSirene.latitude];
      }
      return null;
    },
    isEntrepreneurIndividuel() {
      if (this.resultSirene.unite_legale.categorie_juridique == "1000") {
        return true;
      }
      return false;
    },
    linkBodacc() {
      return "https://www.bodacc.fr/annonce/liste/" + this.resultSirene.siren;
    }
  }
};
</script>

<style lang="scss" scoped>
.company__main {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

#map {
  margin-left: 2em;
}

@media (max-width: $desktop) {
  .company__main {
    flex-direction: column;
  }

  #map {
    margin-left: 0;
  }
}

.company__buttons {
  margin-top: 1.5em;

  .button {
    padding: 0.5em 1em 0.6em;
    vertical-align: middle;
    margin-left: 0;
  }
}

h2 {
  margin: 0;
}

.subtitle {
  font-size: 1.25em;
}

.second__subtitle {
  margin-top: 0.5em;
}

.company__siren,
.second__subtitle {
  color: $color-darker-grey;
}

.map__dummy {
  height: 350px;
  width: 48%;
  background-color: #f2eae2;
}

.link__external {
  margin-top: 10px;
  display: inline-block;
}
</style>
