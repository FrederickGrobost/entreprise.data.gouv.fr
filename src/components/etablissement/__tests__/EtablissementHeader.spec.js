import { createLocalVue, shallowMount, RouterLinkStub } from '@vue/test-utils'
import { __createMocks as createStoreMocks } from '@/store/index.js'
import Vuex from 'vuex'
import EtablissementHeader from '@/components/etablissement/EtablissementHeader.vue'

const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('@/store/index.js')
jest.mock('mapbox-gl', () => ({
  supported: jest.fn(),
  Popup: jest.fn()
}))

describe('EtablissementHeader.vue Snapshot testing', () => {
  const storeMocks = createStoreMocks()

  const wrapperEH = shallowMount(EtablissementHeader, {
    localVue,
    store: storeMocks.store,
    stubs: {
      RouterLink: RouterLinkStub
    }
  })

  test('It match the snapshot', () => {
    expect(wrapperEH.vm.$el).toMatchSnapshot()
  })
})
