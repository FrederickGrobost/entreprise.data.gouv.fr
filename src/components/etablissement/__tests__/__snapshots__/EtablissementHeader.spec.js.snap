// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EtablissementHeader.vue Snapshot testing It match the snapshot 1`] = `
<div
  class="company"
>
  <div
    class="company__main"
  >
    <div
      class="title__block"
    >
      <h2>
        
        
        
        <span
          class="company__siren"
        >
          (
          <span />
          )
        </span>
      </h2>
       
      <!---->
       
      <div
        class="subtitle"
      >
        <div>
          
        </div>
         
        <div>
          
        </div>
      </div>
       
      <div
        class="second__subtitle"
      >
        
          
        
      </div>
       
      <!---->
       
      <etablissement-sirene-children-stub />
       
      <a>
        
        Fiche d'immatriculation au RNCS
      
      </a>
       
      <etablissement-header-timestamp-stub
        resultrna="[object Object]"
        resultsirene="[object Object]"
      />
    </div>
     
    <etablissement-map-stub
      coordinates=""
      etablissement="[object Object]"
    />
  </div>
</div>
`;
